components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/AuthorizedAPIDAOImpl.java

```diff
@@ -235,17 +179,277 @@ public AuthorizedAPI getAuthorizedAPI(String applicationId, String apiId, int te
+
+        this.addAuthorizedAPI(applicationId, authorizedAPI.getAPIId(), authorizedAPI.getPolicyId(),
+                authorizedAPI.getScopes(), authorizedAPI.getAuthorizationDetailsTypes(), tenantId);
+    }
+
+    @Override
+    public void patchAuthorizedAPI(String appId, String apiId, List<String> scopesToAdd, List<String> scopesToRemove,
+                                   List<String> authorizationDetailsTypesToAdd,
+                                   List<String> authorizationDetailsTypesToRemove, int tenantId)
+            throws IdentityApplicationManagementException {
+
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try {
+                this.addScopes(dbConnection, appId, apiId, scopesToAdd, tenantId);
+
+                this.deleteScopes(dbConnection, appId, apiId, scopesToRemove, tenantId);
+
+                this.addAuthorizedAuthorizationDetailsTypes(dbConnection, appId, apiId,
+                        authorizationDetailsTypesToAdd, tenantId);
+
+                this.deleteAuthorizedAuthorizationDetailsTypes(dbConnection, appId, apiId,
+                        authorizationDetailsTypesToRemove, tenantId);
+
+                IdentityDatabaseUtil.commitTransaction(dbConnection);
+            } catch (SQLException | APIResourceMgtException e) {
+                IdentityDatabaseUtil.rollbackTransaction(dbConnection);
+                throw e;
+            }
+        } catch (SQLException | APIResourceMgtException e) {
+            throw new IdentityApplicationManagementException("Error while updating the authorized API. Caused by, ", e);
+        }
+    }
+
+    @Override
+    public List<AuthorizationDetailsType> getAuthorizedAuthorizationDetailsTypes(String applicationId, int tenantId)
+            throws IdentityApplicationManagementException {
+
+        List<AuthorizationDetailsType> authorizationDetailsTypes = new ArrayList<>();
+
+        if (isRichAuthorizationRequestsDisabled()) {
+            return authorizationDetailsTypes;
+        }
+
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement prepStmt = dbConnection
+                     .prepareStatement(ApplicationMgtDBQueries.GET_AUTHORIZED_AUTHORIZATION_DETAILS_TYPES)) {
+
+            prepStmt.setString(1, applicationId);
+            ResultSet resultSet = prepStmt.executeQuery();
+
+            while (resultSet.next()) {
+                authorizationDetailsTypes.add(this.buildAuthorizationDetailsTypeWithSchema(resultSet));
+            }
+
+            return authorizationDetailsTypes;
+        } catch (SQLException e) {
+            throw new IdentityApplicationManagementException("Error while retrieving " +
+                    "authorized authorization details types. Caused by, ", e);
+        }
+    }
+
+    private void addAuthorizedAPI(String applicationId, String apiId, String policyId, List<Scope> scopes,
```

Comment:
lets add doc comments

Reply:
other methods as well

Reply:
Fixed in 2d72c264576eb9e95bb40ef8fef70f8cdd5cbeb1